### PR TITLE
cli: remove unnecessary command line flags

### LINKED ISSUE #4754: cli: Remove unnecessary command line flags
Let's remove command line flags which operators will not have any use for changing. Easier to add these later if they become necessary; harder to remove.

```
--memtable-budget
--metrics-frequency
--scan-interval
--scan-max-idle-time
--time-until-store-dead
```

### PR DESCRIPTION
**Pull request by BramGruneir**
fixes #4754 

attn: @jseldess

---

This change is [Review on Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/5033)

### CODE REVIEW

File: server/context.go [hunk 1]
```diff
@@ -249,6 +263,42 @@ func (ctx *Context) InitNode() error {
...
+		if linearizable, err := strconv.ParseBool(linearizableString); err != nil {
+			log.Errorf("could not parse environment variable cockroach-linearizable=%s, setting to default of %t, error: %s",
+				linearizableString, ctx.Linearizable, err)
+		} else {
+			ctx.Linearizable = linearizable
+			log.Infof("\"linearizable\" set to %t based on cockroach-linearizable environment variable", ctx.Linearizable)
+		}
+	}
+
+	parseDurationEnv("cockroach-max-offset", "max offset", &ctx.MaxOffset)
```
Comment (petermattis) [inline server/context.go]: Environment variables are almost always all caps and separated with underscores. While this isn't required, people will look at us funny if we use lowercase env variables.

Comment (BramGruneir) [inline server/context.go]: I'll change them all up.  Expect and updated PR in a few mins.

Comment (BramGruneir) [inline server/context.go]: done

File: server/context_test.go [hunk 2]
```diff
@@ -72,3 +74,103 @@ func TestParseJoinUsingAddrs(t *testing.T) {
...
 }
+
+// TestReadEnvironmentVariables verifies that all environment variables are
+// correctly parsed.
+func TestReadEnvironmentVariables(t *testing.T) {
+	defer leaktest.AfterTest(t)()
+
+	resetEnvVar := func() {
+		// Reset all environment variables in case any were already set.
+		if err := os.Unsetenv("COCKROACH-LINEARIZABLE"); err != nil {
```
Comment (petermattis) [inline server/context_test.go]: `s/-/_/g`

**Comment by petermattis** [comment]
LGTM

File: server/context_test.go (hunk 2 above)
Comment (BramGruneir) [inline server/context_test.go]: bah, fixed.

**Pull request merged**